\documentclass{article}
\usepackage{graphicx} % Allow insertion of graphics.
\usepackage{authblk} % Better layout of affiliations.
\usepackage[margin=1in]{geometry} % Set margins.
\usepackage{xcolor} % Allow colored text.
\usepackage{url} % Allow URLs.
\usepackage[sort&compress]{natbib} % Better bibliography formatting.

\newcommand{\fixme}[1]{{\color{red}{\bf FIXME: #1}\color{black}}}

\title{A multi-species benchmark for training and validating large scale mass spectrometry proteomics machine learning models}

\author[1,2]{William Stafford Noble}

\affil[1]{Department of Genome Sciences, University of Washington}
\affil[2]{Paul G.\ Allen School of Computer Science and Engineering, University of Washington}

\date{}

\begin{document}

\maketitle

\fixme{Instructions at
  \url{https://www.nature.com/sdata/publish/submission-guidelines}.}

\begin{abstract}
  \fixme{Insert abstract here.}
\end{abstract}

\section*{Background \& Summary}

\textit{De novo} sequencing of proteomics tandem mass spectrometry data, in which observed fragmentation spectra are translated into corresponding peptide sequences, has been an open challenge for more than 40 years \cite{sakurai1984paas}.  Recently, as in many other areas of science, considerable progress toward solving this challenge has been made using deep learning, in which multi-layer neural networks with millions of parameters are trained to generate peptide sequences from observed spectra.  The first such deep learning method, DeepNovo \cite{tran2017denovo}, has been followed by at least 22 additional publications (reviewed in \cite{XXX}).

Because deep learning models are large, they typically require a correspondingly large amount of data to train.  \fixme{Mention how many parameters some important models have, and how much data they were trained on.}

\fixme{Talk about complications.  Labeling strategy.  Train/test leakage.  Batch effects.  Quality of the spectra.}



\section*{Methods}

\subsection*{Data sets}

We created a new version of the nine-species benchmark originally described by Tran {\em et al.} \cite{tran2017denovo}

\begin{enumerate}
\item Paiva \textit{et al.}\ investigated the protein expression response of the cowpea plant (\textit{Vigna unguiculata}) to infection by \textit{Cowpsea severe mosaic virus} (CSMV) by carrying out label-free proteomic analysis of cowpea leaves that were inoculated with CSMV compared to mock inoculation controls \cite{paiva2016label}.
\item Nevo \textit{et al.}\ studied a rare autosomal recessive lysosomal storage disorder, cystinosis, by carrying out SILAC proteomic analysis of engineered mouse cell lines that harbor a known pathenogenic mutation of the causative gene, \textit{CTNS} \cite{nevo2017impact}.
\item Cassidy \textit{et al.}\ evaluated two different analytical approaches for carrying out full proteome analysis while identifying short open reading frames: high/low pH reversed phase LC-MS bottom-up approach and a semi-top-down strategy involving separation of proteins a GelFree system followed by digestion and LC-MS analysis \cite{cassidy2016combination}. The experiments were carried out using the methane producing archaeon \textit{Methanosarcina mazei}.
\item Reuss \textit{et al.}\ carried out proteomic analyses on a series of minimized strains of the model bacterium, \textit{Bacillus subtilis}, with genomes reduced by $\sim$36\% \cite{reuss2017large}.
\item Petersen \textit{et al.}\ performed proteomic analysis of \textit{Candidatus endoloripes}, which are bacterial symbionts of the \textit{Lucinidae} family of marine bivalves \cite{petersen2016chemosynthetic}.
\item Mata \textit{et al.}\ characterized the proteome of the tomato pericarp at its ripe red stage \cite{mata2017depth}.
\item Seidel \textit{et al.}\ analyzed the global proteomic stress response in wildtype and two yeast knockout strains for the gene PBP1  \cite{seidel2017quantitative}.
\item Hu \textit{et al.}\ studied honeybees that exhibit a suite of behaviors (\textit{Varroa} sensitive hygiene---VSH) associated with infection with the \textit{Varroa destructor} virus \cite{hu2016proteome}.  Proteomic analysis was carried out on mushroom bodies and antennae of adult honeybees with and without VSH.
\item Cypryck \textit{et al.} characterized extracellular vesicles released from human primary macrophages after infection with influenza A viruses \cite{cypryk2017proteomic}.
\end{enumerate}
All nine studies were performed using a Thermo Scientific Q Exactive mass spectrometer.

% The PXD entry says the species is "Vigna mungo (Rice bean) (Black gram); NCBI TaxID: 3915;" but the paper talks about Vinga unguiculata.


To do so, we downloaded the RAW files from the same nine PRIDE projects (Supplementary Table~\ref{tab:benchmark}) and converted them to MGF format using the ThermoRawFileParser v1.3.4.
We also downloaded the corresponding nine Uniprot reference proteomes and constructed a Tide index for each one, using Crux version 4.1.
For one species (\textit{Vigna mungo}), no reference proteome is available, so we used the proteome of the closely related species \textit{Vigna radiata}.

\subsection*{Assigning peptide labels to spectra}

We specified Cys carbamidomethylation as a static modification and allowed for the following variable modifications: Met oxidation, Asn deamidation, Gln deamidation, N-term acetylation, N-term carbamylation, N-term NH$_{3}$ loss, and the combination of N-term carbamylation and NH$_{3}$ loss by using the tide-index options {\tt --mods-spec 1M+15.994915,\allowbreak 1N+0.984016,\allowbreak 1Q+0.984016 --nterm-peptide-mods-spec 1X+42.010565,\allowbreak 1X+43.005814,\allowbreak 1X-17.026549,\allowbreak 1X+25.980265 --max-mods 3}.
Note that one of the nine experiments (\textit{Mus musculus}) was performed using SILAC labeling, but we searched without SILAC modifications and hence include in the benchmark only PSMs from unlabeled peptides.
Each index also contains a shuffled decoy peptide corresponding to each target peptide.
Each MGF file was searched against the corresponding index using the precursor window size and fragment bin tolerance specified in the original study (Supplementary Table~\ref{tab:benchmark}).
We used XCorr scoring with Tailor calibration \cite{sulimov2020tailor}, and we allowed for 1 isotope error in the selection of candidate peptides.
All search results were then analyzed jointly per species using the Crux implementation of Percolator, with default parameters.
For the benchmark, we retained all PSMs with Percolator q value $<$ 0.01.
We identified 13 MGF files with fewer than 100 accepted PSMs, and we eliminated all of these PSMs from the benchmark.

\subsection*{Avoiding train/test leakage}

We then post-processed the PSMs to eliminate peptides that are shared between species.
Among the 229,984 unique peptides, we identified 3797 (1.7\%) that occur in more than one species.
For each such peptide, we selected one of the associated species at random and then eliminated all PSMs containing that peptide in other species.
Note that when identifying shared peptides between species, we
considered all modified forms of a given peptide sequence to be the
same.  Hence, if a given peptide appears in more than one species,
then that peptide, including all its modified forms, is randomly
assigned to a single species and eliminated from the others.
The final benchmark dataset consists of 2.8 million PSMs drawn from 343 RAW files.

\subsection*{Balancing the benchmark}

At this stage, the benchmark is quite imbalanced, in the sense that some species have a much larger number of associated PSMs.  We therefore used a random downsampling procedure to produce a benchmark that is more evenly balanced across species.  Among the nine species, the one with the fewest PSMs is \textit{Mus musculus}, with 25,541.  Downsampling all of the other eight species to have 25,000 PSMs would reduce the size of the dataset from 2.8 million PSMs to 225,000---a reduction of 92\%.  To avoid producing such a small dataset, we therefore opted to downsample each dataset to approximately 100,000 PSMs.  This approach yields a slight imbalance, because three species have fewer than 100,000 PSMs (44,604 for \textit{H.\ sapiens} and 82,290 for \textit{Candidatus endoloripes}), while retaining a larger percentage of the original data.  Our downsampling procedure involved randomly permuting the order of the MGF files for each species and then selecting the files in order until at least 100,000 PSMs have been accepted.  The final, balanced benchmark dataset consists of \fixme{XXX} PSMs.

\section*{Data Records}

The dataset contains files resulting from various stages of the benchmark:
\begin{itemize}
\item Spectrum files in MGF format, produced by msconvert.
\item Reference proteome files in FASTA format, downloaded from Uniprot.
\item Search results files for both targets and decoys, in tab-delimited format, produced by Tide.
\item PSM-level Percolator results files for targets, in tab-delimited format.
\item Annotated MGF files for all three versions of the benchmark (overlap, main, and balanced).
\end{itemize}
Also included are log files for the steps of the analysis pipeline carried out using Crux \cite{park2008rapid} (Tide indexing, Tide search, and Percolator).

\section*{Technical Validation}

\fixme{False discovery rate curves}

\section*{Usage Notes (optional)}

\section*{Code Availability}


\section*{References}

\paragraph{Author Contributions}

\paragraph{Competing Interests}
  

\bibliographystyle{unsrt}
\bibliography{refs} % See https://github.com/Noble-Lab/noble-lab-references

\section*{Supplementary information}

\appendix
\renewcommand{\theequation}{S\arabic{equation}}
\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thesection}{S\arabic{section}}
\renewcommand{\thetable}{S\arabic{table}}
\setcounter{table}{0}
\setcounter{figure}{0}


\end{document}

% How to make a dense list.
%\usepackage{enumitem}
%\begin{enumerate}[noitemsep,nolistsep]
